packages/flutter_tools/lib/src/commands/apk.dart

```diff
@@ -0,0 +1,213 @@
+        defaultsTo: _kDefaultOutputPath,
+        help: 'Output APK file.');
+    argParser.addOption('target',
+        abbr: 't',
+        defaultsTo: '',
+        help: 'Target app path or filename used to build the FLX..');
+    argParser.addOption('flx',
+        abbr: 'f',
+        defaultsTo: '',
+        help: 'Path to the FLX file. If this is not provided, an FLX will be built.');
+  }
+
+  int _buildApk(BuildConfiguration config, String flxPath) {
+    File androidManifest = new File(argResults['manifest']);
+    File icuData = new File('${runner.enginePath}/third_party/icu/android/icudtl.dat');
+    File classesDex = new File('${config.buildDir}/gen/sky/shell/shell/classes.dex');
+    File libSkyShell = new File('${config.buildDir}/gen/sky/shell/shell/shell/libs/armeabi-v7a/libsky_shell.so');
+    File keystore = new File('${runner.enginePath}/build/android/ant/chromium-debug.keystore');
+
+    for (File f in [androidManifest, icuData, classesDex, libSkyShell, keystore]) {
+      if (!f.existsSync()) {
+        _logging.severe('Can not locate file: ${f.path}');
+        return 1;
+      }
+    }
+
+    Directory androidSdk = new Directory('${runner.enginePath}/third_party/android_tools/sdk');
+    if (!androidSdk.existsSync()) {
+      _logging.severe('Can not locate Android SDK: ${androidSdk.path}');
+      return 1;
+    }
+
+    Directory tempDir = Directory.systemTemp.createTempSync('flutter_tools');
+    try {      
+      _AssetBuilder assetBuilder = new _AssetBuilder(tempDir, 'assets');
+      assetBuilder.add(icuData, 'icudtl.dat');
+      assetBuilder.add(new File(flxPath), 'app.flx');
+
+      _AssetBuilder artifactBuilder = new _AssetBuilder(tempDir, 'artifacts');
+      artifactBuilder.add(classesDex, 'classes.dex');
+      artifactBuilder.add(libSkyShell, 'lib/armeabi-v7a/libsky_shell.so');
+
+      _ApkBuilder builder = new _ApkBuilder(androidSdk.path);
+      File unalignedApk = new File('${tempDir.path}/app.apk.unaligned');
+      builder.package(unalignedApk, androidManifest, assetBuilder.directory,
+                      artifactBuilder.directory);
+      builder.sign(keystore, _kKeystorePassword, _kKeystoreKeyName, unalignedApk);
+
+      File finalApk = new File(argResults['output-file']);
+      _ensureDirectoryExists(finalApk.path);
+      builder.align(unalignedApk, finalApk);
+
+      return 0;
+    } finally {
+      tempDir.deleteSync(recursive: true);
+    }
+  }
+
+  @override
+  Future<int> runInProject() async {
+    if (runner.enginePath == null) {
+      _logging.severe('Unable to locate the Flutter engine.  Use the --engine-src-path option.');
+      return 1;
+    }
+
+    BuildConfiguration config = buildConfigurations.firstWhere(
+        (BuildConfiguration bc) => bc.targetPlatform == TargetPlatform.android);
+
+    String flxPath = argResults['flx'];
+
+    if (!flxPath.isEmpty) {
+      if (!FileSystemEntity.isFileSync(flxPath)) {
+        _logging.severe('FLX does not exist: $flxPath');
```

Comment:
should this mention that you don't have to specify an flx, it'll be created for you if you don't set it?